src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -69,6 +84,146 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+    void ProfileViewModel::UpdateFontList() noexcept
+    try
+    {
+        // initialize font list
+        std::vector<hstring> fontList;
+        std::vector<hstring> monospaceFontList;
+
+        // get a DWriteFactory
+        com_ptr<IDWriteFactory> factory;
+        THROW_IF_FAILED(DWriteCreateFactory(
+            DWRITE_FACTORY_TYPE_SHARED,
+            __uuidof(IDWriteFactory),
+            reinterpret_cast<::IUnknown**>(factory.put())));
+
+        // get the font collection; subscribe to updates
+        com_ptr<IDWriteFontCollection> fontCollection;
+        THROW_IF_FAILED(factory->GetSystemFontCollection(fontCollection.put(), TRUE));
+
+        for (UINT32 i = 0; i < fontCollection->GetFontFamilyCount(); ++i)
+        {
+            try
+            {
+                // get the font family
+                com_ptr<IDWriteFontFamily> fontFamily;
+                THROW_IF_FAILED(fontCollection->GetFontFamily(i, fontFamily.put()));
+
+                // get the font's localized names
+                com_ptr<IDWriteLocalizedStrings> localizedFamilyNames;
+                THROW_IF_FAILED(fontFamily->GetFamilyNames(localizedFamilyNames.put()));
+
+                // use our current locale to find the correct name
+                BOOL exists{ FALSE };
+                UINT32 index;
+                HRESULT hr;
+                wchar_t localeName[LOCALE_NAME_MAX_LENGTH];
+                if (GetUserDefaultLocaleName(localeName, LOCALE_NAME_MAX_LENGTH))
+                {
+                    hr = localizedFamilyNames->FindLocaleName(localeName, &index, &exists);
+                }
+                if (SUCCEEDED(hr) && !exists)
+                {
+                    hr = localizedFamilyNames->FindLocaleName(L"en-us", &index, &exists);
+                }
+                if (!exists)
+                {
+                    // failed to find the correct locale, using the first one
+                    index = 0;
+                }
+
+                // get the localized name
+                UINT32 nameLength;
+                THROW_IF_FAILED(localizedFamilyNames->GetStringLength(index, &nameLength));
+
+                std::wstring localizedName;
+                localizedName.resize(nameLength);
+                THROW_IF_FAILED(localizedFamilyNames->GetString(index, localizedName.data(), nameLength + 1));
+
+                // get the standard version of that font
+                com_ptr<IDWriteFont> font;
+                THROW_IF_FAILED(fontFamily->GetFirstMatchingFont(DWRITE_FONT_WEIGHT::DWRITE_FONT_WEIGHT_NORMAL,
+                                                                 DWRITE_FONT_STRETCH::DWRITE_FONT_STRETCH_NORMAL,
+                                                                 DWRITE_FONT_STYLE::DWRITE_FONT_STYLE_NORMAL,
+                                                                 font.put()));
+
+                // add the font name to our list
+                if (font.as<IDWriteFont1>()->IsMonospacedFont())
```

Comment:
This throws if it fails to QI, right? This just LOOKS like a null deref potential at quick glance.

Reply:
yea, `as<IFoo>` will throw if it's not a `IFoo`, `try_as<IFoo>` will return `nullptr`